cadc-vosi/src/main/java/ca/nrc/cadc/vosi/CapGetAction.java

```diff
@@ -144,13 +143,33 @@ public void doAction() throws Exception {
         logInfo.setSuccess(true);
     }
     
-    private void transformAccessURL(Capabilities caps, String hostname) throws MalformedURLException {
-        for (Capability cap : caps.getCapabilities()) {
-            for (Interface i : cap.getInterfaces()) {
+    // transform all accessURL so the hostname and context path match that used to invoke
+    // the /capabilities endpoint
+    private void transform(Capabilities caps) throws MalformedURLException {
+        log.debug("context: " + syncInput.getContextPath());
+        log.debug("component: " + syncInput.getComponentPath());
+        
+        String hostname = new URL(syncInput.getRequestURI()).getHost();
+        
+        // find context path in the template using capabilities endpoint
+        Capability cap = caps.findCapability(Standards.VOSI_CAPABILITIES);
+        URL capURL = cap.getInterfaces().get(0).getAccessURL().getURL();
+        String capPath = capURL.getPath();
+        String basePath = capURL.getPath().substring(0, capPath.indexOf("/capabilities")); // chop
+        
+        // capabilities in the request
+        String actualPath = syncInput.getContextPath();
+        
+        log.debug("transform: basePath in template: " + basePath + " actualPath: " + actualPath);
+        for (Capability c : caps.getCapabilities()) {
+            for (Interface i : c.getInterfaces()) {
                 AccessURL u = i.getAccessURL();
                 URL url = u.getURL();
-                URL nurl = new URL(url.getProtocol(), hostname, url.getPath());
+                String path = url.getPath();
+                String npath = path.replace(basePath, actualPath);
+                URL nurl = new URL(url.getProtocol(), hostname, npath);
                 u.setURL(nurl);
+                log.debug("transform: " + url + " -> " + nurl);
             }
         }
     }
```

Comment:
I think this should all be fine, considering the actualPath currently always contains the name of the war file.